homepage/models.py

```diff
@@ -40,6 +40,65 @@ def get_details(self):
         return (self.course_id, self.name, self.mandatory, self.credit_points, self.syllabi,
                 self.avg_load, self.avg_rating, self.num_of_raters, self.num_of_reviewers)
 
+    # --- get all courses sorted by id/name/rating  - return QuerySets
+    @staticmethod
+    def get_all_courses_sorted_by_id():
+        return Course.objects.all()
+
+    @staticmethod
+    def get_all_courses_sorted_by_name():
+        courses = Course.objects.order_by('name')
+        return courses
+
+    @staticmethod
+    def get_all_courses_sorted_by_rating():
+        courses = Course.objects.order_by('avg_rating')
+        return courses[::-1]
+
+    # --- course lists sorting by id/name/rating  - return lists
+    @staticmethod
+    def course_list_sort_by_name(course_list):
+        # gets a list of course object instances and sorts it by name
+        course_list.sort(key=lambda course: course.name)
+        return course_list
+
+    @staticmethod
+    def course_list_sort_by_id(course_list):
+        # gets a list of course object instances and sorts it by course id
+        course_list.sort(key=lambda course: course.course_id)
+        return course_list
+
+    @staticmethod
+    def course_list_sort_by_rating(course_list):
+        # gets a list of course object instances and sorts it by rating
+        course_list.sort(key=lambda course: course.rating)
+        return course_list[::-1]
+
+    # --- get filtered course list by rating/load/mandatory/elective  - return QuerySets
+    @staticmethod
+    def get_filtered_courses_by_rating(rating):
+        # gets all courses with average rating >= rating
+        courses = Course.objects.filter(avg_rating__gte=rating)
+        return courses
+
+    @staticmethod
+    def get_filtered_courses_by_load(load):
+        # gets all courses with average course load <= load
+        courses = Course.objects.filter(avg_load__lte=load)
+        return courses
+
+    @staticmethod
+    def get_mandatory_courses():
+        # gets all courses that are mandatory
+        courses = Course.objects.filter(mandatory=True)
+        return courses
+
+    @staticmethod
+    def get_elective_courses():

```

Comment:
Not a big difference, but maybe you can combine this method with ```get_mandatory_courses```,
and pass ```mandatory``` as parameter

Reply:
I like the separation, myself, because I can't think of a name other than "get_mandatory_yay_nay" to describe a method that does both depending on input